src/js/components/deployment-view/index.tsx

```diff
@@ -50,20 +52,40 @@ class DeploymentView extends React.Component<Props, void> {
   }
 
   public componentWillMount() {
-    const { loadPreviewAndComments, isUserLoggedIn, match: { params: { entityType, token, id } } } = this.props;
+    const {
+      loadPreviewAndComments,
+      isUserLoggedIn,
+      isLoggingIn,
+      match: {
+        params: {
+          entityType,
+          token,
+          id,
+        },
+      },
+    } = this.props;
 
     if (!isEntityType(entityType)) {
       console.error('Unknown preview type!');
       return;
     }
 
-    loadPreviewAndComments(id, entityType, token, isUserLoggedIn);
+    if (
+      isUserLoggedIn ||
+      (!isUserLoggedIn && !isLoggingIn) // A non-Minard user viewing an open deployment
```

Comment:
Since the Deployment View could technically be open to non-logged-in users, we have to render it in order to try to fetch the preview information. Only if it fails can we know whether the user should be logged in or not.

It's easy to conditionally render components based on logged in status, though.